src/credentials/providers/tempCredentialsProvider.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Credentials } from '@aws-sdk/types'
+import { CredentialType } from '../../shared/telemetry/telemetry.gen'
+import { getStringHash } from '../../shared/utilities/textUtilities'
+import { CredentialsId, CredentialsProvider, CredentialsProviderType } from './credentials'
+import { StaticProfile } from '../types'
+
+/**
+ * A credentials provider for a temporary use case.
+ *
+ * This provides the bare minimum way to use credentials
+ * in the rest of the system.
+ *
+ * @see CredentialsProviderType
+ */
+export class TempCredentialProvider implements CredentialsProvider {
```

Comment:
This is a lot of code for what looks like a hack(?). If this is not intended for use anywhere except the `addTempCredential` hack, the docstring here should make that very clear. Perhaps better would be to define this subclass in the same module as `addTempCredential` so that it's not exported.